### DISCUSSION TITLE
Evaluating results from SIwave Induced Voltage simulation

### DISCUSSION THREAD

**Discussion by SteffenSee92** [ansys/pyedb#705] [Q&A]
Hi everyone,

I have a PCB layout in SIwave, where I performed an Induced Voltage simulation and managed up to this part using pyEDB and some of the old IronPython function from the SIwave scripting guide.

Now, I want to analyze the results from that simulation also by scripting. In SIwave, I would simply click on the results, then choose for example "Plot magnitude" and the magnitudes of the voltages measured at the ports are plotted in Electronics Desktop over frequency. I am interested also in these voltage levels, but found no way to pull these values into my script. Is there a way to get the results from within the SIwave script (without need of working in the GUI)?

An alternative would be to read the values from a text based file in the siwaveresults folder. But unfortunately, there is no such file, I can easily extract the values from. A readable file (.ivswp)  is only being created when clicking on the results in the GUI, so that the plotting in AEDT is made. Is there any way to generate this ivswp file by scripting?

Thanks, best regards
Steffen

**Comment by kmahajan-cadfem** [comment]
@svandenb-dev, could you please let us know if the ivswp files can be generated or if we can access the values directly in python.

**Reply by Samuelopez-ansys** [reply]
Hi @SteffenSee92 and @kmahajan-cadfem ,

This method is still not implemented. Please submit an Issue and the community will consider to implement it. PyEDB is an open source code, so you could also join the community, here you can find the SIwave part:

https://github.com/ansys/pyedb/blob/main/src/pyedb/siwave.py

As you can see there are some implemented parts, like _run_icepak_simulation_, _run_dc_simulation_.... So, if you know the Ironpython commands, you could perform your first contribution to this repository! If you do not have time, or the skills to do it, no problem :) Just create the issue, and the community will do it. You could just convert this discussion to an issue (right column "Create issue from discussion).

Regards,
Samuel

**Reply by kmahajan-cadfem** [reply]
Hi @Samuelopez-ansys, thank you for the answer and also explanation, we have yet to find the IronPython command for this, we will have a look again in the documentation, if this exists, maybe we can do a workaround first by calling directly the IronPython methods and then a PR at later point.

**Reply by svandenb-dev** [reply]
@SteffenSee92 @kmahajan-cadfem 
sorry for the delay I just came back at the office. I would recommend to avoid using the Siwave ironpython API for several reasons. This API is limited and Ansys development is recommended using EDB based workflow. 
I would define ready to solve project with simulation setup, probes.... everything is available in PyEDB (if something is missing please fill an Issue in GitHub). Once the EDB saved you can simply call Aedt line submission with passing the EDB as argument, this will automaticity creates the Aedt file. Once it is solved you can open the project with PyAEDT and access all post processing quantities. I think this would be the easiest way. If you want to use Siwave API you can access from Siwave class:

from pyaedt.siwave import Siwave
siwave = Siwave(specified_version=None)  #optional argument
oDoc = siwave.oproject
oDoc.ScrImportEDB(<folderPath>)
